libbeat/beat/beat.go

```diff
@@ -141,65 +124,114 @@ func init() {
 	rand.Seed(seed)
 }
 
-// newInstance creates and initializes a new Beat instance.
-func newInstance(name string, version string, bt Beater) *instance {
-	if version == "" {
-		version = defaultBeatVersion
+// Run initializes and runs a Beater implementation. name is the name of the
+// Beat (e.g. packetbeat or metricbeat). version is version number of the Beater
+// implementation. bt is the `Creator` callback for creating a new beater
+// instance.
+func Run(name, version string, bt Creator) error {
+	return handleError(newBeat(name, version).launch(bt))
+}
+
+// newBeat creates a new beat instance
+func newBeat(name, version string) *Beat {
+	return &Beat{
+		Name:    name,
+		Version: version,
```

Comment:
For our beats, `version` is empty, so I think this should check for empty and then set version to `defaultBeatVersion`. At startup our beats seem to be printing `Version:` but no number.